lib/libutee/include/pta_generic.h

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Intel Corporation
+ */
+
+#ifndef __PTA_GENERIC_H__
+#define __PTA_GENERIC_H__
+
+#define PTA_GENERIC_UUID { 0xfeca9a1d, 0x5ff0, 0x4204, { \
+			  0xb4, 0x83, 0x34, 0x86, 0x23, 0x71, 0x2d, 0xc9 } }
+
+#define OPTEE_MRC_GENERIC_OPEN			1
+#define OPTEE_MRC_GENERIC_CLOSE			2
+#define OPTEE_MRC_GENERIC_SERVICE_START		3
```

Comment:
Why are start and stop needed? Where's invoke?

Reply:
`InvokeCommand` is to be negotiated between TA and CA, similar to `hello_world/ta/include/hello_world_ta.h`. Since, this is custom service, `optee_os` will not be knowing what commands will be there. PTA will just pass the commands to REE service using tee-supplicant and will get error for non-existent commands.

Reply:
Why cannot start be implicit by open and stop be implicit by close?

Reply:
Somewhere I posted with a typo. Correcting here. Apologies for creating a bit of confusion.

The following are to be written by a custom service.
```
OPTEE_MRC_GENERIC_SERVICE_START
OPTEE_MRC_GENERIC_SERVICE_STOP
```

The following is handled by `tee-supplicant`
```
OPTEE_MRC_GENERIC_OPEN (sends in the UUID to find the custom REE service)
```
find the relevant REE service (based on Message Queue/Dynamic Library) and corresponding `OPTEE_MRC_GENERIC_CLOSE` closes message queue/Dynamic Library

Reply:
I'm sorry, I can see the answer to my question.

Reply:
I will rephrase it. Let me know where I missed to address the point.

`OPTEE_MRC_REE_SERVICE_OPEN/CLOSE` is used by `tee-supplicant` to open/close a message queue or to get/put a handle to DLL.

`OPTEE_MRC_REE_SERVICE_START/STOP` is to be implemented by REE service. REE service can setup itself before it starts receiving the next command.

`InvokeCommand()` uses the above defines in `commandID`.

Reply:
So why cannot the REE Service automatically do START when it has been received a connection from tee-supplicant and do STOP when the connection is teared down. I don't see why this has to be driven by secure world.

Reply:
Let's start with the first command sent `OPTEE_MRC_REE_SERVICE_OPEN`. It sends following parameters to Normal World.

o param[0] = OPTEE_MRC_REE_SERVICE_OPEN
o param[1]= uuid
o param[2] = handle to filled by tee-supplicant.

There is no space for passing any TA specific parameters. So, a new command is sent internally which is `START` and it is intended to pass any parameters which TA requested.